### PR TITLE
Introduce ExchangeType

### LINKED ISSUE #90: ExchangeDeclareArguments: exchange type could be an enum
There is a small number of built-in exchange types plus any arbitrary `x-*` custom types.

By implementing `From` string and string slices, this can probably be a be source-compatible change.

### PR DESCRIPTION
**Pull request by michaelklishin**
and adapt ExchangeDeclareArguments to support it in a backwards-compatible way.

Closes #90.

### CODE REVIEW

**Comment by michaelklishin** [comment]
I'll take a look at the Clippy warning:

```
error: an implementation of `From` is preferred since it gives you `Into<_>` for free where the reverse isn't true
```

File: amqprs/src/api/channel/exchange.rs [hunk 1]
```diff
@@ -8,14 +10,105 @@ use super::{Channel, Result};
 #[cfg(feature = "compliance_assert")]
 use crate::api::compliance_asserts::assert_exchange_name;
 
+/// Exchange types. Most variants are for exchange types included with modern RabbitMQ distributions.
+/// For custom types provided by 3rd party plugins, use the `Plugin(String` variant.
```
Comment (gftea) [inline amqprs/src/api/channel/exchange.rs:14]:
```suggestion
/// For custom types provided by 3rd party plugins, use the `Plugin(String)` variant.
```

**Comment by gftea** [comment]
Thanks for bring in knowledge of the ecosystem and the improvements. Because we are aiming test coverage above 85%, please add test cases for the ExchangeType

**Comment by michaelklishin** [comment]
@gftea done

**Review by gftea — APPROVED** [review]

**Comment by gftea** [comment]
thanks a lot @michaelklishin !

**Pull request merged**